src/ucstring.rs

```diff
@@ -770,9 +789,8 @@ impl UCString<u32> {
     /// assert!(res.is_err());
     /// assert_eq!(res.err().unwrap().nul_position(), 1);
     /// ```
-    pub fn from_chars(v: impl Into<Vec<char>>) -> Result<Self, NulError<u32>> {
-        let v: Vec<u32> = unsafe { mem::transmute(v.into()) };
-        UCString::new(v)
+    pub fn from_chars(v: Vec<char>) -> Result<Self, NulError<u32>> {
+        UCString::new(v.into_iter().map(u32::from).collect::<Vec<_>>())
```

Comment:
Won't this new implementation cause the vec to get reallocated instead of merely reinterpreted? This could be a perf regression.

Maybe a better implementation, if the goal is to weed out the transmutes, to do something along the lines of

```suggestion
        let ptr = v.as_mut_ptr(); let len = v.len(); let capacity = v.capacity();
        std::mem::forget(v);
        UCString::new(Vec::from_raw_ptr(ptr as *mut u32, len, capacity))
```

(Ideally, we'd use https://doc.rust-lang.org/alloc/vec/struct.Vec.html?search=#method.into_raw_parts, but that API is nightly-only still :( )